--!strict

export type ErrorCode = {
    ACTION_COOLDOWN: string,
    ACTION_RATE_LIMITED: string,
    SCHEMA_ARRAY_REQUIRED: string,
    SCHEMA_CFRAME_REQUIRED: string,
    SCHEMA_COLOR3_REQUIRED: string,
    SCHEMA_ENUM_ITEM_REQUIRED: string,
    SCHEMA_INSTANCE_ANCESTRY: string,
    SCHEMA_INSTANCE_CLASS: string,
    SCHEMA_INSTANCE_REQUIRED: string,
    SCHEMA_INTEGER_REQUIRED: string,
    SCHEMA_INTERNAL_TYPE: string,
    SCHEMA_NUMBER_OUT_OF_RANGE: string,
    SCHEMA_STRING_TOO_LONG: string,
    SCHEMA_STRING_TOO_SHORT: string,
    SCHEMA_TYPE_MISMATCH: string,
    SCHEMA_UNION_MISMATCH: string,
    SCHEMA_UNKNOWN_FIELD: string,
    SCHEMA_VECTOR3_REQUIRED: string,
}

local ErrorCode = table.freeze({
    ACTION_COOLDOWN = "ACTION_COOLDOWN",
    ACTION_RATE_LIMITED = "ACTION_RATE_LIMITED",
    SCHEMA_ARRAY_REQUIRED = "SCHEMA_ARRAY_REQUIRED",
    SCHEMA_CFRAME_REQUIRED = "SCHEMA_CFRAME_REQUIRED",
    SCHEMA_COLOR3_REQUIRED = "SCHEMA_COLOR3_REQUIRED",
    SCHEMA_ENUM_ITEM_REQUIRED = "SCHEMA_ENUM_ITEM_REQUIRED",
    SCHEMA_INSTANCE_ANCESTRY = "SCHEMA_INSTANCE_ANCESTRY",
    SCHEMA_INSTANCE_CLASS = "SCHEMA_INSTANCE_CLASS",
    SCHEMA_INSTANCE_REQUIRED = "SCHEMA_INSTANCE_REQUIRED",
    SCHEMA_INTEGER_REQUIRED = "SCHEMA_INTEGER_REQUIRED",
    SCHEMA_INTERNAL_TYPE = "SCHEMA_INTERNAL_TYPE",
    SCHEMA_NUMBER_OUT_OF_RANGE = "SCHEMA_NUMBER_OUT_OF_RANGE",
    SCHEMA_STRING_TOO_LONG = "SCHEMA_STRING_TOO_LONG",
    SCHEMA_STRING_TOO_SHORT = "SCHEMA_STRING_TOO_SHORT",
    SCHEMA_TYPE_MISMATCH = "SCHEMA_TYPE_MISMATCH",
    SCHEMA_UNION_MISMATCH = "SCHEMA_UNION_MISMATCH",
    SCHEMA_UNKNOWN_FIELD = "SCHEMA_UNKNOWN_FIELD",
    SCHEMA_VECTOR3_REQUIRED = "SCHEMA_VECTOR3_REQUIRED",
}) :: ErrorCode

return ErrorCode
